### ISSUE TITLE
Replace transmute_copy in Hash implementation for ApplicableDeclarationsCacheQuery

### ISSUE THREAD

**Issue by jdm** [servo/servo#5446]
This looks like it's just transmuting a pointer into an integer. We should be able to cast `&Vec<DeclarationBlock>` to `*const Vec<DeclarationBlock>`, then cast that to a usize value without using any unsafe code.

**Comment by Ms2ger** [comment]
That's not quite what it does. The code is equivalent to
```rust
let x = declaration as *const DeclarationBlock as *const uint;
let ptr = unsafe { *x };
ptr.hash(state);
```
or
```rust
let ptr = &*declaration.declarations as *const Vec<PropertyDeclaration> as uint;
ptr.hash(state);
```

**Comment by jdm** [comment]
Whoops, yeah.

**Comment by aneeshusa** [comment]
New to Rust here. Is there any reason we can't do the following?
```rust
let ptr = declaration as *const DeclarationBlock as usize;
ptr.hash(state);
```
This seems fine based on [the only reference I could find on type casting] (https://github.com/rust-lang/rfcs/blob/master/text/0401-coercions.md#casts), and this compiles fine for me.

**Comment by Ms2ger** [comment]
It is something you can do, yes. It is not clear to me what the code is supposed to do, though, and so I can't tell if this is the correct behaviour.

The code dates back to 0fa0940ce9f043d19036894dc4349f9a4b2767aa; @pcwalton, so you remember what you were trying to do?

**Comment by eefriedman** [comment]
This was fixed in #7523.
